targets/particle/source/main.cpp

```diff
@@ -0,0 +1,149 @@
+          const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  if (args_cnt != 2)
+  {
+    Serial.println ("Wrong arguments count in 'test.setLed' function.");
+    return jerry_create_boolean (false);
+  }
+
+  int ledPin = jerry_get_number_value (args_p[0]);
+  bool value = jerry_get_boolean_value (args_p[1]);
+
+  pinMode (ledPin, OUTPUT);
+  digitalWrite (ledPin, value);
+
+  return jerry_create_boolean (true);
+} /* set_led */
+
+/**
+ * Delay function
+ */
+static jerry_value_t
+js_delay (const jerry_value_t func_value, /**< function object */
+          const jerry_value_t this_value, /**< this arg */
+          const jerry_value_t *args_p, /**< function arguments */
+          const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  if (args_cnt != 1)
+  {
+    Serial.println ("Wrong arguments count in 'test.delay' function.");
+    return jerry_create_boolean (false);
+  }
+
+  int millisec = jerry_get_number_value (args_p[0]);
+
+  delay (millisec);
+
+  return jerry_create_boolean (true);
+} /* js_delay */
+
+/*
+ * Init available js functions
+ */
+static void
+init_jerry ()
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  /* Create an empty JS object */
+  jerry_value_t object = jerry_create_object ();
+
+  jerry_value_t func_obj;
+  jerry_value_t prop_name;
+
+  func_obj = jerry_create_external_function (set_led);
+  prop_name = jerry_create_string ((const jerry_char_t *) "setLed");
+  jerry_set_property (object, prop_name, func_obj);
+  jerry_release_value (prop_name);
+  jerry_release_value (func_obj);
+
+  func_obj = jerry_create_external_function (js_delay);
+  prop_name = jerry_create_string ((const jerry_char_t *) "delay");
+  jerry_set_property (object, prop_name, func_obj);
+  jerry_release_value (prop_name);
+  jerry_release_value (func_obj);
+
+  /* Wrap the JS object (not empty anymore) into a jerry api value */
+  jerry_value_t global_object = jerry_get_global_object ();
+
+  /* Add the JS object to the global context */
+  prop_name = jerry_create_string ((const jerry_char_t *) "test");
+  jerry_set_property (global_object, prop_name, object);
+  jerry_release_value (prop_name);
+  jerry_release_value (object);
+  jerry_release_value (global_object);
+} /* init_jerry */
+
+/**
+ * Jerryscript simple test
+ */
+static void
+test_jerry ()
+{
+  const jerry_char_t script[] = " \
+    test.setLed(7, true); \
+    test.delay(250); \
+    test.setLed(7, false); \
+    test.delay(250); \
+  ";
+
+  size_t script_size = strlen ((const char *) script);
+
+  jerry_value_t eval_ret = jerry_eval (script, script_size, false);
+
+  /* Free JavaScript value, returned by eval */
+  jerry_release_value (eval_ret);
+} /* test_jerry */
+
+/**
+ * Setup code for particle firmware
+ */
+void
+setup ()
+{
+  Serial.begin (9600);
+  delay (2000);
+  Serial.println ("Beginning Listening mode test!");
+} /* setup */
+
+/**
+ * Loop code for particle firmware
+ */
+void
+loop ()
+{
+  init_jerry ();
```

Comment:
Is jerry initialized and cleaned in every loop iteration? Does particle/photon application model have no concept of initialization and cleanup, so that the loop could consist of test_jerry only?